dataactbroker/handlers/fileHandler.py

```diff
@@ -642,3 +727,17 @@ def getProtectedFiles(self):
 
         response["urls"] = self.s3manager.getFileUrls(bucket_name=CONFIG_BROKER["static_files_bucket"], path=CONFIG_BROKER["help_files_path"])
         return JsonResponse.create(StatusCode.OK, response)
+
+    def completeGeneration(self, generationId):
+        """ Retrieve the generated file, using the specified ID to look up submission and file type """
+        if generationId is None:
+            return JsonResponse.error(ResponseException("Must include a generation ID",StatusCode.CLIENT_ERROR), StatusCode.CLIENT_ERROR)
+
+        # Pull url from request
+        safeDictionary = RequestDictionary(self.request)
+        url =  safeDictionary.getValue("href")
+        #Pull information based on task key
+        task = self.interfaces.jobDb.session.query(FileGenerationTask).options(joinedload(FileGenerationTask.file_type)).filter(FileGenerationTask.generation_task_key == generationId).one()
```

Comment:
Note that this'll blow up if the submission id isn't in our database. Should it 400 instead?